### PR TITLE
Fix parsing v100s in `get_available_flops`

### LINKED ISSUE #18946: FLOPs not found for 'Tesla V100S-PCIE-32GB'
### Bug description

```sh
Seed set to 1337
Seed set to 1337
/workspace/.miniconda3/lib/python3.11/site-packages/lightning/fabric/utilities/throughput.py:349: FLOPs not found for 'Tesla V100S-PCIE-32GB'
```

however in the code [seems to exist](https://github.com/Lightning-AI/lightning/blob/4af77c6bf17bf868c8c649bc415fba679906bbf4/src/lightning/fabric/utilities/throughput.py#L340).

### What version are you seeing the problem on?

v2.1

### How to reproduce the bug

```python
Finetuning with lit-gpt


python finetune/lora.py \
  --data_dir data/alpaca \
  --checkpoint_dir checkpoints/togethercomputer/RedPajama-INCITE-Base-3B-v1 \
  --out_dir out/lora_weights/RedPajama-INCITE-Base-3B-v1 \
  --precision 16-true --quantize bnb.nf4
```

### Error messages and logs

```sh
Seed set to 1337
Seed set to 1337
/workspace/.miniconda3/lib/python3.11/site-packages/lightning/fabric/utilities/throughput.py:349: FLOPs not found for 'Tesla V100S-PCIE-32GB'
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

### More info

_No response_

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #18946

----
:books: Documentation preview :books:: https://pytorch-lightning--18952.org.readthedocs.build/en/18952/

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

File: tests/tests_fabric/utilities/test_throughput.py [hunk 1]
```diff
@@ -61,6 +61,29 @@ def test_available_flops(xla_available):
...
+        "a10g",
+        "V100-sxm",
+        "v100-pcie",
+        "v100s-pcie",
+        "t4",
+        "quadro rtx 5000",
+    ],
+)
+@mock.patch("lightning.fabric.accelerators.cuda._is_ampere_or_later", return_value=False)
+def test_get_available_flops_cuda_mapping_exists(_, device_name):
```
Comment (awaelchli) [inline tests/tests_fabric/utilities/test_throughput.py:81]: @carmocca Let me know if you have any suggestions to improve the test. For now, it just tests that we can parse all known entries.

**Review by Borda — APPROVED** [review]

File: tests/tests_fabric/utilities/test_throughput.py [hunk 2]
```diff
@@ -61,6 +61,29 @@ def test_available_flops(xla_available):
     tpu.reset_mock()
 
 
+@pytest.mark.parametrize(
+    "device_name",
+    [
+        # TODO: We need to represent the real names here
+        "h100-hbm3",
+        "h100-pcie",
```
Comment (carmocca) [inline tests/tests_fabric/utilities/test_throughput.py:69]:
```suggestion
        "NVIDIA H100 PCIe",
```

File: tests/tests_fabric/utilities/test_throughput.py [hunk 3]
```diff
@@ -61,6 +61,29 @@ def test_available_flops(xla_available):
...
+        "h100-hbm2e",
+        "a100",
+        "a10g",
```
Comment (carmocca) [inline tests/tests_fabric/utilities/test_throughput.py:72]:
```suggestion
        "NVIDIA A10G",
```

**Comment by carmocca** [comment]
Changing the milestone since this is unreleased

**Pull request merged**